adapters/AppLovin/AppLovinAdapter/GADMAppLovinMediatedNativeAppInstallAd.m

```diff
@@ -0,0 +1,116 @@
+//
+//  GADMAppLovinMediatedNativeAppInstallAd.m
+//  SDK Network Adapters Test App
+//
+//  Created by Santosh Bagadi on 4/12/18.
+//  Copyright © 2018 AppLovin Corp. All rights reserved.
+//
+
+#import "GADMAppLovinMediatedNativeAppInstallAd.h"
+
+#import "GADMAppLovinExtraAssets.h"
+
+@interface GADMAppLovinMediatedNativeAppInstallAd() <GADMediatedNativeAdDelegate>
+
+@property(nonatomic, strong) ALNativeAd *nativeAd;
+@property(nonatomic, strong) NSArray *nativeAdImages;
+@property(nonatomic, strong) GADNativeAdImage *nativeAdIcon;
+@property(nonatomic, copy) NSDictionary<NSString *, id> *extras;
+
+@end
+
+@implementation GADMAppLovinMediatedNativeAppInstallAd
+
+- (instancetype)initWithNativeAd:(ALNativeAd *)nativeAd {
+    if (!nativeAd)
+    {
+        return nil;
+    }
+    
+    self = [super init];
+    if (self) {
+        self.nativeAd = nativeAd;
+        UIImage *downloadedImage =
```

Comment:
Should take GADNativeAdImageAdLoaderOptions  into account before downloading the images. Please refer to the [documentation.](https://developers.google.com/admob/ios/api/reference/Classes/GADNativeAdImageAdLoaderOptions#/c:objc(cs)GADNativeAdImageAdLoaderOptions(py)disableImageLoading)

Reply:
With regards to this - our SDK already pre-caches the assets before returning the `ALNativeAd` to the adapter.